roipoly/roipoly.py

```diff
@@ -69,15 +69,18 @@ def __init__(self, fig=None, ax=None, color='b',
         self.__cid2 = self.fig.canvas.mpl_connect(
             'button_press_event', self.__button_press_callback)
 
+        # determine the type of environment
+        using_spyder = any(['SPYDER' in x for x in os.environ])
```

Comment:
Also, rather define this as a constant (this piece of code should only be executed once, upon module loading. So just put `USING_SPYDER` directly after the imports.

Reply:
Thanks, I think a lot of users have problems with spyder, so nice that someone takes care of that!

On which platfoms have you run the tests? MacOS, Win, Linux?

Reply:
> What kind of env variables do you expect here? Better to be explicit if we know it, rather than checking if any env variable name contains the string `SPYDER` .

Spyder has the following environmental variables (on Windows at least with spyder 4.0.2): 
```python
['SPYDER_ARGS', 'SPY_EXTERNAL_INTERPRETER', 'SPY_UMR_ENABLED', 'SPY_UMR_VERBOSE', 
'SPY_UMR_NAMELIST', 'SPY_RUN_LINES_O', 'SPY_PYLAB_O', 'SPY_BACKEND_O', 
'SPY_AUTOLOAD_PYLAB_O', 'SPY_FORMAT_O', 'SPY_BBOX_INCHES_O', 'SPY_RESOLUTION_O', 
'SPY_WIDTH_O', 'SPY_HEIGHT_O', 'SPY_USE_FILE_O', 'SPY_RUN_FILE_O', 'SPY_AUTOCALL_O', 
'SPY_GREEDY_O', 'SPY_JEDI_O', 'SPY_SYMPY_O', 'SPY_TESTING', 'SPY_HIDE_CMD']
```
I'm not sure there's any bullet-proof way of doing this as a user could set their own variables and mess with any ad hoc approach we come up with. Perhaps counting the number of varaibles that start with `SPY_` and if there are more than `n` then detect spyder? This may all fall down for different version of spyder.

> And, what if someone has installed SPYDER, but is not running roipoly from spyder, is the env variable then also set? Or only, if someone launches roipoly from the spyder IDE?

Variables containing 'SPY' do not exist (by default) when running python with spyder installed.

> On which platfoms have you run the tests? MacOS, Win, Linux?

I have only tested this on Windows as I only have access to spyder on a windows IDE.

FYI: I have no experience of using spyder aside from trying to dissuade from using it.

Reply:
How about checking for these four variables?
``` 
    spyder_args = os.environ.pop('SPYDER_ARGS', None)
    pid = os.environ.pop('SPYDER_PID', None)
    is_bootstrap = os.environ.pop('SPYDER_IS_BOOTSTRAP', None)
    reset = os.environ.pop('SPYDER_RESET', None)
``` 
from https://github.com/spyder-ide/spyder/blob/master/spyder/app/restart.py#L189

Reply:
Updated. I guess we can't do better than what SPYDER do themselves.